src/shared/use-is-compulsory-data-element-operand.js

```diff
@@ -0,0 +1,57 @@
+import { useMemo } from 'react'
+import { useMetadata, selectors } from './metadata/index.js'
+import { useDataSetId } from './use-context-selection/use-context-selection.js'
+import { useDataValueSet } from './use-data-value-set/use-data-value-set.js'
+
+export const useIsCompulsoryDataElementOperand = ({
```

Comment:
I colocated these hooks because I didn't really want to create a new folder in src/shared directory. I'd also vote for moving the miscellaneous hooks in src/shared into src/shared/hooks, and then I could create a folder and two separate hooks for these, but I didn't want to create two separate files with the the existing file structure.